src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -301,9 +305,21 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
         }
       });
 
+    // detect when select all checkbox changes
+    this.selectAllCheckbox.valueChanges
+      .pipe(takeUntil(this._onDestroy))
+      .subscribe(() => {
+        this.emitSelectAllBooleanToParent();
+      });
+
     this.initMultipleHandling();
   }
 
```

Comment:
```suggestion
_isToggleAllCheckboxVisible(): boolean {
  return this.matSelect.multiple && this.showToggleAllCheckbox;
}
```

Reply:
this subscription is needed to send the value of the checkbox to the parent component when it changes over time. I have implemented your suggested function of _isToggleAllCheckboxVisible() in my latest commit, but have kept the subscription calling the emit function.